### PR TITLE
Feature/validate sponsor

### PR DESCRIPTION
**Pull request by fdurand**
# Description
Enable or no if a sponsor need to authenticate when he click on the link to validate the access.

# Impacts
No

# Delete branch after merge
YES

# NEWS file entries

## Enhancements
* Enable or disable Sponsor authentication when he validate access.

### CODE REVIEW

**Comment by extrafu** [comment]
@julsemaan Review and merge if all right.

**Comment by julsemaan** [comment]
IMO, it would be risky to merge feature PRs given our tests are already underway

Unless we explicitly assign someone to retest sponsor once that is merged

**Comment by extrafu** [comment]
Ok let's push it to v8.1.

**Comment by fdurand** [comment]
BUMP

File: html/pfappserver/lib/pfappserver/Form/Config/Source/SponsorEmail.pm [hunk 1]
```diff
@@ -61,6 +61,17 @@ has_field 'sponsorship_bcc' => (
     },
 );
 
+has_field 'validate_sponsor' =>
+  (
+   type => 'Toggle',
+   checkbox_value => 'yes',
+   unchecked_value => 'no',
+   label => 'Sponsor validation',
```
Comment (cgx) [inline html/pfappserver/lib/pfappserver/Form/Config/Source/SponsorEmail.pm]: Please capitalize all words of the label.

File: html/pfappserver/lib/pfappserver/Form/Config/Source/SponsorEmail.pm [hunk 2]
```diff
@@ -61,6 +61,17 @@ has_field 'sponsorship_bcc' => (
...
+   default => pf::Authentication::Source::SponsorEmailSource->meta->get_attribute('validate_sponsor')->default,
+   tags => { after_element => \&help,
+             help => 'Do you want to authenticate the sponsor when he click on the link ?' },
```
Comment (cgx) [inline html/pfappserver/lib/pfappserver/Form/Config/Source/SponsorEmail.pm]: Don't use the interrogative form. I suggest the following:

> Force sponsor to authenticate when validating a guest request.

**Comment by atran-inverse** [comment]
I got this error after signing in as the sponsor:
Caught exception in captiveportal::Controller::Activate::Email->doSponsorRegistration "No authentication context provided at /usr/local/pf/lib/pf/authentication.pm line 299."

**Comment by fdurand** [comment]
@atran-inverse , it looks to be related to  that issue https://github.com/inverse-inc/packetfence/issues/3250 can you try it ?

**Comment by atran-inverse** [comment]
Feature is tested and working.
Merging.

**Pull request merged**